hubble/src/scroll.rs

```diff
@@ -111,21 +111,18 @@ impl Scroll {
         let storage = self
             .l1_client
             .get_storage_at(
-                ethers::types::H160(self.rollup_finalization_config.rollup_contract_address.0),
-                H256::from(
+                alloy::primitives::Address::new(
+                    (self.rollup_finalization_config.rollup_contract_address).0,
+                ),
+                alloy::primitives::Uint::from_be_bytes(
                     self.rollup_finalization_config
                         .rollup_last_finalized_batch_index_slot
                         .to_be_bytes(),
-                )
-                .into(),
-                Some(ethers::types::BlockId::Number(
-                    ethers::types::BlockNumber::Number((l1_height as u64).into()),
-                )),
+                ),
             )
             .await
             .wrap_err("error fetching l1 rollup contract storage")?;
-
-        let batch_index = U256::from_be_bytes(storage.to_fixed_bytes())
+        let batch_index: u64 = storage
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/unionlabs/union/pull/2693#issuecomment-2293655644>#ef6b49</a><span> - 08/16/2024, 03:05 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The conversion from 'storage' to 'u64' using 'try_into' may fail if 'storage' is not a valid 'u64'. This can lead to a panic at runtime.
</div>
<div id="fix">
<b>Fix:</b> Implement proper error handling for the conversion from 'storage' to 'u64' to avoid potential runtime panics.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -125,3 +125,7 @@
 +        let batch_index: u64 = match storage.try_into() {
 +            Ok(value) => value,
 +            Err(_) => return Err(eyre::eyre!("Failed to convert storage to u64")),
 +        };
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged